### PR TITLE
relax restmapper resource matching

### PR DESCRIPTION
**Pull request by deads2k**
We were matching case insensitive on Kinds, not Resources, thus driving me insane.

@liggitt @caesarxuchao

### CODE REVIEW

**Comment by caesarxuchao** [comment]
LGTM, but will let @liggitt take another pass.

File: pkg/api/meta/restmapper.go [hunk 1]
```diff
@@ -199,7 +200,19 @@ func (m *DefaultRESTMapper) ResourceSingularizer(resourceType string) (string, e
 	return singular.Resource, nil
 }
 
-func (m *DefaultRESTMapper) ResourcesFor(resource unversioned.GroupVersionResource) ([]unversioned.GroupVersionResource, error) {
+// coerceResourceForMatching makes the resource lower case and converts internal versions to unspecified (legacy behavior)
+func coerceResourceForMatching(input unversioned.GroupVersionResource) unversioned.GroupVersionResource {
+	resource := input.GroupVersion().WithResource(strings.ToLower(input.Resource))
```
Comment (liggitt) [inline pkg/api/meta/restmapper.go]: why not `input.Resource = strings.ToLower(input.Resource)`?

Comment (deads2k) [inline pkg/api/meta/restmapper.go]: > why not input.Resource = strings.ToLower(input.Resource)? we're already getting a struct copy when we call this func... not sure why we'd want to create two more (intermediate GroupVersion and new GroupVersionResource)

Personal distaste for mutating args, but I don't care a whole lot

Comment (deads2k) [inline pkg/api/meta/restmapper.go]: done

**Comment by liggitt** [comment]
Lgtm

**Pull request merged**